### PR TITLE
BaSP-MR-3: admins put and delete

### PR DESCRIPTION
**Pull request by Roberto-Orazi**
I finish everyting and try it in postman and works correctly

### CODE REVIEW

File: src/models/Admin.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    lastName: {
+      type: String,
+      required: true,
+      minLength: 3,
+      maxLength: 15,
+    },
+    dni: {
+      type: Number,
+      required: true,
+      minLength: 7,
```
Comment (AleMac17) [inline src/models/Admin.js:21]: minLength is not a validator for numbers in mongoose
you have to use min

File: src/models/Admin.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      maxLength: 8,
```
Comment (AleMac17) [inline src/models/Admin.js:22]: maxLength is not a validator for numbers in mongoose
You have to use max

File: src/models/Admin.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    },
+    phone: {
+      type: Number,
+      required: true,
+      minLength: 10,
```
Comment (AleMac17) [inline src/models/Admin.js:27]: minLength is not a validator for numbers in mongoose
You have to use min

File: src/validations/admins.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const Joi = require('joi');
+
+const validateUpdateAdmin = (req, res, next) => {
+  const validateAdmin = Joi.object({
+    firstName: Joi.string().min(3).max(15),
+    lastName: Joi.string().min(3).max(15),
+    dni: Joi.number().min(10000000).max(99999999),
```
Comment (AleMac17) [inline src/validations/admins.js:7]: In the schema you put this 
` dni: {
      type: Number,
      required: true,
      minLength: 7,
      maxLength: 8,
    },`
    In the dni validation here you put numbers with 8 digits but in the schema you put 7 digits

**Review by AleMac17 — CHANGES_REQUESTED** [review]

File: src/models/Admin.js [hunk 5]
```diff
@@ -35,7 +35,7 @@ const adminSchema = new Schema(
     password: {
       type: String,
       required: true,
-      minLength: 6,
+      min: 6,
```
Comment (AleMac17) [inline src/models/Admin.js:38]: In Mongoose min in numbers its not the same as minlength. You have to put the exact min number
example `min: 100000`

File: src/models/Admin.js [hunk 6]
```diff
@@ -6,25 +6,25 @@ const adminSchema = new Schema(
...
-      minLength: 7,
-      maxLength: 8,
+      min: 7,
+      max: 8,
     },
     phone: {
       type: Number,
       required: true,
-      minLength: 10,
+      min: 10,
```
Comment (AleMac17) [inline src/models/Admin.js:27]: In Mongoose min in numbers its not the same as minlength. You have to put the exact min number
example `min: 1000000000`

File: src/models/Admin.js [hunk 7]
```diff
@@ -6,25 +6,25 @@ const adminSchema = new Schema(
...
-      maxLength: 15,
+      min: 3,
+      max: 15,
     },
     dni: {
       type: Number,
       required: true,
-      minLength: 7,
-      maxLength: 8,
+      min: 7,
```
Comment (AleMac17) [inline src/models/Admin.js:21]: In Mongoose min in numbers its not the same as minlength. You have to put the exact min number
example `min: 1000000`

File: src/models/Admin.js [hunk 8]
```diff
@@ -6,25 +6,25 @@ const adminSchema = new Schema(
...
+      max: 8,
```
Comment (AleMac17) [inline src/models/Admin.js:22]: In Mongoose max in numbers its not the same as maxlength. You have to put the exact max number
example `max: 99999999`

**Review by AleMac17 — CHANGES_REQUESTED** [review]

File: src/models/Admin.js [hunk 9]
```diff
@@ -6,25 +6,25 @@ const adminSchema = new Schema(
...
-      maxLength: 15,
+      min: 3,
+      max: 15,
     },
     lastName: {
       type: String,
       required: true,
-      minLength: 3,
-      maxLength: 15,
+      min: 3,
```
Comment (AleMac17) [inline src/models/Admin.js:15]: For strings you need to use minLength and maxLength

File: src/models/Admin.js [hunk 10]
```diff
@@ -6,25 +6,25 @@ const adminSchema = new Schema(
     firstName: {
       type: String,
       required: true,
-      minLength: 3,
-      maxLength: 15,
+      min: 3,
```
Comment (AleMac17) [inline src/models/Admin.js:9]: For strings you need to use minLength and maxLength

**Review by AleMac17 — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+      });
+    }
+    return res.status(200).json({
+      message: 'Admin was deleted succesfully',
+      data: foundAdmin,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Server error.',
```
Comment (AleMac17) [inline src/controllers/admins.js:62]: In the message of the catch return, we need to put an error.
for example ` message: error,`

File: src/controllers/admins.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+export const deleteAdmin = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const foundAdmin = await Admins.findByIdAndDelete(id);
+    if (!foundAdmin) {
+      return res.status(404).json({
+        message: `${id} was not found`,
+        data: [{}],
+        error: false,
```
Comment (AleMac17) [inline src/controllers/admins.js:52]: Status 404 is considered an error.
should be: `error: true,`

File: src/controllers/admins.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+      },
+      {
+        new: true,
+      },
+    );
+    if (!foundAdmin) {
+      return res.status(404).json({
+        message: `${id} was not found`,
+        data: [{}],
+        error: false,
```
Comment (AleMac17) [inline src/controllers/admins.js:27]: Status 404 is considered an error.
should be: `error: true,`

File: src/controllers/admins.js [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+    return res.status(200).json({
+      message: 'Admin was updated succesfully',
+      data: foundAdmin,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Server error',
+      data: error,
+      error,
```
Comment (AleMac17) [inline src/controllers/admins.js:39]: The: true is missing after the error.
should be: `error: true,`

File: src/controllers/admins.js [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    return res.status(200).json({
+      message: 'Admin was updated succesfully',
+      data: foundAdmin,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Server error',
+      data: error,
```
Comment (AleMac17) [inline src/controllers/admins.js:38]: In data we are using one of these.  data: {} || [] || undefined

File: src/controllers/admins.js [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+      });
+    }
+    return res.status(200).json({
+      message: 'Admin was updated succesfully',
+      data: foundAdmin,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Server error',
```
Comment (AleMac17) [inline src/controllers/admins.js:37]: In the message of the catch return, we need to put an error.
for example  `message: error,`

**Review by AleMac17 — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+        password,
+      },
+      {
+        new: true,
+      },
+    );
+    if (!foundAdmin) {
+      return res.status(404).json({
+        message: `${id} was not found`,
+        data: [{}],
```
Comment (AleMac17) [inline src/controllers/admins.js:26]: In data we are using only one of these: data: {} || [] || undefined

File: src/controllers/admins.js [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+export const deleteAdmin = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const foundAdmin = await Admins.findByIdAndDelete(id);
+    if (!foundAdmin) {
+      return res.status(404).json({
+        message: `${id} was not found`,
+        data: [{}],
```
Comment (AleMac17) [inline src/controllers/admins.js:51]: In data we are using only one of these: data: {} || [] || undefined

**Review by AleMac17 — CHANGES_REQUESTED** [review]

**Review by AleMac17 — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**